src/core/regimes.rkt

```diff
@@ -25,16 +25,42 @@
 ;; `infer-splitpoints` and `combine-alts` are split so the mainloop
 ;; can insert a timeline break between them.
 
-(define (infer-splitpoints alts ctx)
-  (timeline-event! 'regimes)
+(define (infer-splitpoints alts branch-exprs cerrs cbest-index ctx)
```

Comment:
I kind of think a bunch of the code would be more readable if `branch-exprs` was a hash table instead of a list. In a couple of places you have to track indices where you already have the branch expr.